apps/dashboard/app/models/files.rb

```diff
@@ -132,12 +132,24 @@ def self.mime_type_for_preview(type)
     #     x === 'application/x-yaml'
     #     # => true
     #
-    if %r{text/.*}.match(type).nil?
-      type
-    else
-      'text/plain; charset=utf-8'
-    end
-  end
+
+    # NOTE:  due to how most modern browsers are handling the yaml mime-types,
+    # we have to cast the yaml mime-types to text/plain mime-type so we can
+    # view the yaml files inline with the browser.
+    # Would like to have a pattern match for the  yaml mime-types; something like application/*yml and application/*yaml
+    # but couldn't find one that worked. Therefore, for now we're using full mime-type names for yaml mime-types.
+
+    castToTextArray = ['text/*','application/x-yml','application/x-yaml','application/yml','application/yaml']
+
+    castToTextArray.each { |mime| 
+      if ! %r{#{mime}}.match(type).nil?
+        type = 'text/plain; charset=utf-8'
+      end
+    }
+
+    type
```

Comment:
For multi-line blocks you want to you want to use the `do end` type notation. 

```ruby
foo.each do |ele|
  # the block
end
```
But I got a one liner for ya

```suggestion
    type = 'text/plain; charset=utf-8' if plain_cast_array.any? { |mime| /#{mime}/.match?(type) }

    type
```

I'll have to check the rubocop `Style/RegexpLiteral` - but looking at it now I think I do prefer slashes.